docs: basic tutorials

Exirel opened the pull request:
### Description

This starts a new set of tutorials for Plugin Author:

* "Your first plugin" introduce the very concept of a Plugin
* "Playing with your command" talks about commands and triggers
* "Configuration and plugin setup" adds the plugin config section

Combined, they provide a first introduction to Sopel Plugin development. They don't talk about everything, trying to introduce very few notions at the same time.

Further tutorials would be welcome in future release of Sopel.

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

dgw commented on docs/source/tutorials/first-plugin.rst:
```diff
@@ -0,0 +1,49 @@
+
+.. seealso::
+
+    To interract with the list of plugins installed, read the documentation
```
```suggestion
    To interact with the list of plugins installed, read the documentation
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+    def shrug(bot, trigger):
+        bot.say('¯\\_(ツ)_/¯')
+
+Which is one of the maintainers' favorite command to use. However, let's see
```
```suggestion
Which is one of the maintainers' favorite commands to use. However, let's see
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+        bot.say('¯\\_(ツ)_/¯')
+
+Which is one of the maintainers' favorite command to use. However, let's see
+if we can do a bit more complex than that.
```
```suggestion
if we can do something a bit more complex than that.
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+Greeting a user by their name
+=============================
+
+Have you noticed that a :func:`plugin_callable` takes two arguments? The first
```
I'd xref this to <tt>:ref:`plugin-anatomy-callables`</tt> to 1) avoid the monospace format and 2) render as a friendly term instead of what looks like a function name.

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+Playing with your command
+=========================
+
+Now that you have started :doc:`first-plugin`, maybe you want to write a more
```
We should consider that links to `:doc:` refs might look better with custom titles, at least to avoid the uppercase first letter that flows weirdly in the middle of a sentence.

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+Command with arguments
+======================
+
+The trigger object can do much more for you: if a user add arguments to the
```
```suggestion
The trigger object can do much more for you: if a user adds arguments to the
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+======================
+
+The trigger object can do much more for you: if a user add arguments to the
+command, like this: ``.hello morning``, you can detect and use that argument::
```
```suggestion
command, like ``.hello morning``, you can detect and use that argument::
```

This is for flow. Colon followed by comma is a bit odd.

(Would be nice if it also puts `.hello morning` on the same line, but I doubt deleting `this: ` will free up enough space in Furo.)

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+And... action!
+==============
+
+Some users say ``.hello`` out loud, and other will say it with an action. How
```
```suggestion
Some users say ``.hello`` out loud, and others will say it with an action. How
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+
+Sopel knows how to register the same plugin callable for different types of
+trigger, so both ``.shrug`` and ``/me shrugs`` work. For example, you could do
+that for your hello plugin::
```
```suggestion
this for your hello plugin::
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+    <Sopel> Hello YourNick, have a nice day!
+
+
+Summing It Up
```
```suggestion
Summing it up
```

Consistency, schmistency.

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+
+    <YourNick> .hello
+    <Sopel> Hello YourNick, have a nice day!
+    *YourNick hi*
```
Is this the action format for any actual clients? Usually I see something like this:

```suggestion
    * YourNick hi
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,129 @@
+=============
+
+In this tutorial, we talked briefly about ``bot.say()`` and ``bot.reply()``,
+and there are more ways to :doc:`interact with the bot </plugin/bot>`.
```
Yep, this is the link style I think all the `:doc:` refs should use.

But also,

```suggestion
and explored a few more ways to :doc:`interact with the bot </plugin/bot>`.
```

dgw commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,106 @@
+
+    from sopel.config import types
+
+    class MyFirstPlugin(types.StaticSection):
```
Decision point: Do we reinforce the core convention of naming the class `PluginNameSection` in this tutorial?

Even if we decide to promote a different conventional name format, it's probably wise to guide the user toward some meaningful suffix, like how most (built-in/stdlib) exceptions in Python end with `Error`.

Exirel replied:
Good point! I'll try to use the same plugin name in the tutorial so it's easier to keep track of that. I believe having consistent naming will be enough for the tutorial.

dgw commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,106 @@
+        fruit = random.choice(bot.settings.firstplugin.fruits)
+        bot.say(f'I want a {fruit}!')
+
+And whenever someone triggers this command, the bot will say that it wants one
+of the configured fruits. If you want to list 50 fruits or only 2 is up to you,
+and to the bot owners who will install your plugin.
```
Any interest in making a joke about the hard-coded `a`? Like "(or only fruits that don't start with a vowel)"? 😁

(Future tutorial including [`inflect`](https://pypi.org/project/inflect/)?)

dgw commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,106 @@
+Putting everything together
+===========================
+
+We can regroup everything together in one file::
```
Further from the phrase shape used in the heading, so it doesn't sound as redundant.

```suggestion
We can combine all of this into one plugin file::
```

Yeah, nitpicks like this _are_ why I never get anything done at $work. Why do you ask?

dgw commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,106 @@
+        bot.say(f'I want a {fruit}!')
+
+As you can see, there are several steps when it comes to configuration. Sopel
+tries to make it as straightforward as possible for you to declare and to setup
```
```suggestion
tries to make it as straightforward and flexible as possible for you to declare and to setup
```

Too much? (I know, it'll need a rewrap.)

dgw commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,106 @@
+your plugin configuration, and it offers various possibilities.
+
+You can read more about :ref:`plugin configuration <plugin-anatomy-config>`,
+which includes a section about the configuration wizzard as well. You can also
```
```suggestion
which includes a section about the configuration wizard as well. You can also
```

Exirel replied:
Wizzzzzzzzzzard!

dgw reviewed (commented):
Who would have expected this many line notes in such a small PR? Well, probably half of them are just slaying the nemesi**S**, so it's not _that_ bad. 😛

Exirel commented:
@dgw I added an introduction in the tutorial root page, because I felt that something was missing. I wanted to tell our reader that 1. they need to know a bit of Python, and 2. writing a plugin is doable, but not all plugin are easy to write!

Apart from this addition, I standardized the plugin name (`myplugin.py`, `MyPluginSection`), and use custom label for `:doc:` links.

I also added a few sentences here and there. In particular, something about the usage of regex. I don't want to overload the reader with too much information about regexes, but a few lines felt better than nothing.

dgw commented on docs/source/tutorials.rst:
```diff
@@ -2,9 +2,36 @@
+too can write a plugin.
+
+These tutorials will guide and help you to begin your journey as a plugin
+author, i.e. someone who can writes plugins for Sopel. Not every plugin is
```
```suggestion
author, i.e. someone who can write plugins for Sopel. Not every plugin is
```

(or "someone who writes", pick your favorite)

dgw commented on docs/source/tutorials.rst:
```diff
@@ -2,9 +2,36 @@
+* to install and run Sopel on your development environnement
+* to have write access to Sopel's configuration and plugin directory
+* a beginner level in Python (e.g. how to write a function, what is a variable,
+  how to perform string formatting, how to access an object's attribute, how
```
```suggestion
  how to perform string formatting, how to access an object's attributes, how
```

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -2,37 +2,39 @@
-need for now is the :attr:`trigger.nick <sopel.trigger.Trigger.nick>`
-attribute::
+For that, you need **the second argument**: the ``trigger``. It is an object
+with all the information you need about the message that triggered the
```
```suggestion
with all the information you need about the message that triggered your
```

Not that the previous wording was _wrong_, but in the context of a tutorial I think "your" makes more sense than "the".

SnoopJ replied:
Adding on to @dgw's suggestion, I think it makes sense to state more directly that the `trigger` holds information about the message, and the rest of the example speaks for the words this edit omits.

```suggestion
with information about the message that triggered your
```

Exirel replied:
This is tough to decide, but ultimately I prefer a more direct approach too.

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -2,37 +2,39 @@
+For that, you need **the second argument**: the ``trigger``. It is an object
+with all the information you need about the message that triggered the
+callable, such as the **message** itself, the **channel**, the type of message,
+etc. and what we need for now is the
```
```suggestion
etc.—and what we need for now is the
```

Getting fancy with that punctuation! (At least, I _hope_ Sphinx can handle this without some kind of escape sequence. It's just Unicode.)

Exirel replied:
Let's get FANCY.
:tophat:
:monocle_face:

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -76,22 +84,35 @@ Now the command will be able to react a bit more to your user::
     <Sopel> Hello YourNick, are you having lunch?
 
+How does that work? Well, the short version is that Sopel uses regex
+(`Regular Expression`__) to match a message to a plugin callable, and the
```
```suggestion
(`REGular EXpressions`__) to match a message to a plugin callable, and the
```

If you can bold-face (**Reg**ular **Ex**pressions) or otherwise emphasize the relevant parts of each word within the link markup, feel free to do that instead; I just don't trust my reST skills enough to suggest that blind. 😁

Exirel replied:
Good call! I'll see what I can do with the **boldening**.

dgw commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -76,22 +84,35 @@ Now the command will be able to react a bit more to your user::
+this for your hello plugin::
 
     @plugin.command('hello')
     @plugin.action_command('hi')
```
```suggestion
    @plugin.action_command('says hi')
```

Don't see any reason this wouldn't work, and it's a better demonstration of how people generally define action commands (i.e. as _actions_ that sound like natural language).

Exirel replied:
Hm... what about this instead?

```suggestion
    @plugin.action_command('waves')
```

dgw reviewed (commented):
Round two complete, I guess. This is shaping up really nicely!

Exirel commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -76,22 +84,35 @@ Now the command will be able to react a bit more to your user::
+.. note::
+
+    In the case of a command, the regex is entirely managed by Sopel itself,
+    while the generic :func:`@plugin.rule <sopel.plugin.rule>` decorator
+    allow you to define your own regex.
```
It's a note in a tutorial: don't worry, it doesn't have to be exhaustive! It is my intent to show only a fraction of what is possible with Sopel here, as to not overload the reader with too many things at once. Take it as a hint, a drop of knowledge put here to open the door and nothing more.

Exirel replied:
Re-thinking about this: I, too, am not happy about how a plugin author can discover all the possible decorators, especially not happy about how to learn to use one or the other. This is definitively something to think about more for a future PR!

Exirel commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -102,24 +123,24 @@ And so, in chat, you will see that::
     <YourNick> .hello
     <Sopel> Hello YourNick, have a nice day!
-    *YourNick hi*
+    * YourNick hi
```
This is how IRCCloud displays the result of a `/me` command:

![image](https://github.com/sopel-irc/sopel/assets/3259552/b02abef7-0b63-46c8-9a57-34141d6a2230)

It doesn't add "says" or anything, because in my experience it is supposed to be a user's action, and not a user talking.

Exirel replied:
```suggestion
    * YourNick waves
```

SnoopJ commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,108 @@
+Declaring your configuration
+============================
+
+To declare a configuration section, you must first create a subclass of
+:class:`~sopel.config.types.StaticSection`, and define attributes::
```
I think it would help to clarify what `define attributes` means here and allude to the fact that there are two different kinds of attributes at play: Python class attributes and Sopel's attribute types for settings. I tried not to get carried away in my suggestion.

```suggestion
To declare a configuration section, you create a subclass of
:class:`~sopel.config.types.StaticSection` that defines a class attribute for
each setting that can be configured in that section. To declare a section with
one list setting named ``fruits``, you can do::

    from sopel.config import types

    class MyPluginSection(types.StaticSection):
        fruits = types.ListAttribute('fruits')

.. note::
    The first ``fruits`` above is the name of this field *on instances of this
    class*, and the second is the name of the section *in the configuration
    file*. It is common (but not required) to use the same name for both.
```

Exirel replied:
This is too much details for a tutorial.

SnoopJ commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,108 @@
+Using your section
+==================
+
+In your configuration file, you can add the following lines:
```
Slight reword to suggest more strongly that this new section means nothing to Sopel without the steps that came before.

```suggestion
Now that you have told Sopel about your custom section, you can add the
following lines in your configuration file:
```

SnoopJ commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,108 @@
+        peach
+        strawberry
+
+And Sopel will expose that for you through ``bot.settings``. For example, you
+can write this command::
```
Slight rewording, emphasize that the section is stored as an instance of the class and use its "full name" in this context (rather than referring to the entire configuration stored in `settings`).

```suggestion
And Sopel will store this section's options in ``bot.settings.myplugin`` using
an instance of the class defined earlier. These settings are now available for
your plugin to use! For example, you can write this command::
```

Exirel replied:
Ditto, too much details. I'll use `bot.settings.myplugin` tho, that's better.

SnoopJ commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,108 @@
+        fruit = random.choice(bot.settings.myplugin.fruits)
+        bot.say(f'I want a {fruit}!')
+
+As you can see, there are several steps when it comes to configuration. Sopel
+tries to make it as straightforward and flexible as possible for you to declare
+and to setup your plugin configuration, and it offers various possibilities.
+
```
I think this paragraph can be dropped entirely, especially since we're about to refer more directly to the `various possibilities` by linking to more thorough documentation :grin: 

```suggestion
```

Exirel replied:
No, it is important to remind the reader that there are multiple steps in the process. It is a warning that, hey, they **must** pay attention to all the requirements. The last bit about "offers various possibilities" could be dropped, reading it now it feels unnecessary.

SnoopJ commented on docs/source/tutorials/first-plugin.rst:
```diff
@@ -0,0 +1,52 @@
+        """Reply with Hello!"""
+        bot.reply('Hello!')
+
+You can put this code in a Python file, placed into your Sopel plugin
+directory, such as ``~/.sopel/plugins/myplugin.py``. Once this is done, you can
```
I think some readers would benefit from explicitly stating here that the bot will load the files it finds in the plugins directory during start-up.

```suggestion
You can put this code in a Python file, placed into your plugins directory,
such as ``~/.sopel/plugins/myplugin.py``, and Sopel will load the plugin when
it scans the directory during start-up. Once this is done, you can
```

Exirel replied:
No, too much details. Right now the tutorial tries to guide the reader to act now and to experiment rather than just reading stuff. It is not the time to make the reader understand how it works, it is the time to make the reader experiment. I want the reader to have the "wahou" effect of "this is easy!", not overload them with too much information.

SnoopJ commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,150 @@
+=============================
+
+Have you noticed that a :ref:`plugin callable <Plugin callables>` takes **two
+arguments?** The first one is the ``bot``, i.e. an instance of Sopel (wrapped
+for your convenience) that you can use to
```
I don't think it's worth mentioning the wrapper implementation detail here, the newbie reader doesn't really need to know about it. I think the `i.e.` can be dropped as well, although I _am_ fond of literary abbreviations (cf. half of what I say on IRC)

```suggestion
arguments?** The first one is the ``bot``, an instance of Sopel that you can
use to
```

dgw replied:
Also we're on track to eliminate the wrapper soon enough, anyway (see #2460).

@SnoopJ's suggestion leaves room for the `:doc:` xref below to be moved onto this line, I'm 98% sure (on mobile rn).

Exirel replied:
Hm... yeah, that makes sense to me.

SnoopJ commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,150 @@
+for your convenience) that you can use to
+:doc:`interact with the bot </plugin/bot>`.
+
+In the previous tutorial, we used ``bot.reply``, which is convenient but not
+always what you want. Maybe you want the bot to say this::
```
Nit: elaborate on what `convenient` is getting at here, the reader may not be familiar enough with what it does

```suggestion
In the previous tutorial, we used ``bot.reply``, which is convenient when             
responding directly to a user, but not always what you want. Maybe you want the       
bot to say something more complex::
```

dgw replied:
I'd say "can be convenient", even, instead of the blanket "is".

SnoopJ commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,150 @@
+.. important::
+
+    If you want to test this with your bot, and your bot is already running,
+    you need to restart the bot so it will load the new version of your plugin.
```
Nit: you don't _need_ to, but we don't _have_ to let them know there's any other way to do it :stuck_out_tongue_winking_eye: 

```suggestion
    restart the bot so it will load the new version of your plugin.
```

dgw replied:
Counter-proposal: Downgrade to "you should" rather than completely removing the prescription.

```suggestion
    you should restart the bot so it will load the new version of your plugin.
```

Exirel replied:
I prefer the more direct approach to avoid a "technical" lie for the greater good.

SnoopJ commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,150 @@
+    @plugin.command('hello')
+    def hello(bot, trigger):
+        """Say Hello <user>, have a nice day!"""
+        # groups 3 to 6 are the first, second, third, and fourth command arg
+        when = trigger.group(3)
```
I think some readers will look at this and think _"huh, what about groups 1 and 2?!"_ so it's probably worth mentioning them even though this example doesn't use them.

An example that would use group 2 could be something like:

```
<YourNick> .shout Hello, is this thing on?
<Sopel> HELLO, IS THIS THING ON?
```

but I think that additional comments will suffice for the careful reader.

```suggestion
        # group 1 is the name of the command that was triggered
        # group 2 is the entire rest of the message
        # groups 3 to 6 are the first, second, third, and fourth command arg
        when = trigger.group(3)
```

Exirel replied:
I went back and forth on that one. Ultimately, it is way too confusing without the mention of what group 1 and 2 are doing.

Exirel replied:
(and no, I'm not happy about it... but that's better that way)

SnoopJ commented on docs/source/tutorials/playing-command.rst:
```diff
@@ -0,0 +1,150 @@
+And... action!
+==============
+
+Some users say ``.hello`` out loud, and others will say it with an action. How
+do you react to these? Let's go back to our initial example of the ``shrug``
+command: it uses a second decorator, ``action_command('shrugs')``, with a
+different name. How does that work?
```
Nit: this command was alluded to, but calling it `our initial example` seems a little bit off to me.

```suggestion
Some users say ``.hello`` out loud, and others will say it with an action. How
do you react to these? Let's go back to the example of the ``shrug`` command:
it uses a second decorator, ``action_command('shrugs')``, with a different
name. How does that work?
```

Exirel replied:
I can see how someone could be confused, so better be safe here. Good call!

SnoopJ reviewed (commented):
Nicely done!

dgw commented on docs/source/tutorials/configurable.rst:
Or `configuration-and-setup`? (More spitballing never hurt, right?)

Exirel replied:
Hm... I like both! I'll say that we are already talking about a tutorial for plugins, so let's go with the "setup" route here.

dgw commented:
> I feel like you both (@dgw and @SnoopJ) want to add more accurate information

If you mean "more _complete_", then we have something to discuss. We _always_ want "accurate" (correct) information in the tutorials, i.e. nothing in the tutorial should be _wrong_ or _misleading_.

If you fixup and resolve the threads about correctness/grammar/phrasing, @SnoopJ and I can then go over the bigger-picture stuff that's left over afterward? (I'm not going to be doing much of this stuff from the highway this week, anyway, I suspect.)

Exirel commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,108 @@
+    def setup(bot):
+        bot.settings.define_section('myplugin', MyPluginSection)
+
+This way, you tell Sopel that the ``[myplugin]`` section is used by your
+plugin, and to parse this section Sopel must use your class, i.e.
+``MyPluginSection``.
```
This is way too much internal behavior explained for a tutorial.

Exirel commented on docs/source/tutorials/configurable.rst:
```diff
@@ -0,0 +1,108 @@
+======================
+
+Now, having a class in your plugin doesn't achieve much: you need to tell the
+bot about it. The best place to do so is in the :func:`setup` function
+hook of your plugin::
```
I'd replace "calling" (which is more accurate) by "using" which is more reader friendly.

Note: you can use this `:meth:~sopel.config.Config.define_section` (with the `~`) to make Sphinx display only the method's name and not the full path.

Exirel replied:
```suggestion
bot about it by using the :meth:`~sopel.config.Config.define_section` method.
The best place to do so is in the :func:`setup` function hook of your plugin::
```

Exirel commented:
> If you mean "more _complete_", then we have something to discuss. 

Did the fix about the thing and now I think you can read the full tutorial and see what's missing while keeping in mind the flow of each chapter.

dgw commented on docs/source/tutorials.rst:
```diff
@@ -0,0 +1,37 @@
+
+These tutorials will guide and help you to begin your journey as a plugin
+author, i.e. someone who can write plugins for Sopel. Not every plugin is
+easy however, and you will probably need to hone your skill with Python, learn
+more about the IRC protocol, and in general learn more about software
+programming. But let's not get ahead of ourselves; you are here for the basics!
```
```suggestion
easy however, and you will probably need to hone your Python skills, learn more
about the IRC protocol, and learn more about software programming in general.
But let's not get ahead of ourselves; you are here for the basics!
```

dgw commented on docs/source/tutorials.rst:
```diff
@@ -0,0 +1,37 @@
+
+Since you'll be running Sopel, we invite you to create a configuration file
+that connects to a friendly IRC server and joins a private testing channel.
+That way, when you restart your bot, or run your command for the hundredth
```
```suggestion
That way, when you restart your bot or run your command for the hundredth
```

I see this as "the hundredth [time]" applying to both restarting & running the same command repeatedly.

(It's a bit difficult to put into words exactly why I didn't like this comma, but that's the best I got.)

Exirel replied:
That's the job of the editor, no question asked. 😁

dgw commented on docs/source/tutorials/configuration-and-setup.rst:
```diff
@@ -0,0 +1,113 @@
+instance of Sopel yourself, you probably had to open and edit its
+:doc:`configuration</configuration>` file.
+
+Usually located in your home directory, under the ``.sopel/`` folder, the
```
```suggestion
Usually located in the ``.sopel/`` folder under your home directory, the
```

I _think_ this is a bit clearer, without going all the way to "`~/.sopel/` folder".

dgw commented on docs/source/tutorials/configuration-and-setup.rst:
```diff
@@ -0,0 +1,113 @@
+Configuration and plugin setup
+==============================
+
+Maybe you :doc:`played with commands </tutorials/playing-command>` for your
```
Broken link. That file was renamed.

```suggestion
Maybe you :doc:`played with commands </tutorials/playing-with-commands>` for your
```

The most recent deploy preview log [flags this](https://app.netlify.com/sites/sopel-unstable-docs/deploys/64e926fbcf9163000742ac0a#L556), but it's hard to find in the sea of other warnings that I believe #2496 will at least reduce:

```
5:12:28 PM: /opt/build/repo/docs/source/tutorials/configuration-and-setup.rst:5: WARNING: unknown document: /tutorials/playing-command
```

dgw commented on docs/source/tutorials/playing-with-commands.rst:
```diff
@@ -0,0 +1,153 @@
+
+Have you noticed that a :ref:`plugin callable <Plugin callables>` takes **two
+arguments?** The first one is the ``bot``, an instance of Sopel that you can
+use to
+:doc:`interact with the bot </plugin/bot>`.
```
```suggestion
use to :doc:`interact with the bot </plugin/bot>`.
```

dgw reviewed (commented):
I didn't go super ham on this because (as you'll see in a line note) I think this is more of a stepping stone than "final" tutorials we won't want to touch for another five years.

Exirel commented on docs/source/tutorials.rst:
```diff
@@ -0,0 +1,37 @@
+=========
+Tutorials
+=========
```
Ha, that's a problem I didn't see, but now it looks obvious to me. You are right, this shouldn't be put there like that. At least the title change is necessary.

As to where to put it... I think we could put that under the plugins for dev section. I'll see what I can do!

dgw commented on docs/source/plugin/tutorials.rst:
```diff
@@ -0,0 +1,37 @@
+
+Before you can dive into these tutorials, you will need the following:
+
+* to install and run Sopel on your development environnement
```
```suggestion
* to install and run Sopel on your development environment
```

🇫🇷🇫🇷🇫🇷

dgw commented on docs/source/plugin/tutorials.rst:
```diff
@@ -0,0 +1,37 @@
+Tutorials
+=========
+
+The key feature of Sopel is its plugin system: everything Sopel does is through
+a plugin. Combining some basic Python knowledge with reading Sopel's
+documentation, you too can write a plugin.
```
```suggestion
The key feature of Sopel is its plugin system: *everything* Sopel does is
through a plugin. Combining some basic Python knowledge with reading Sopel's
documentation, you can write a plugin too!
```

dgw commented on docs/source/plugin/tutorials.rst:
```diff
@@ -0,0 +1,37 @@
+author, i.e. someone who can write plugins for Sopel. Not every plugin is
+easy however, and you will probably need to hone your Python skills, learn more
+about the IRC protocol, and learn more about software programming in general.
+But let's not get ahead of ourselves; you are here for the basics!
```
```suggestion
But let's not get ahead of ourselves; you are here for the basics.
```

I'd rather save the ! for above.

dgw commented on docs/source/plugin/tutorials/playing-with-commands.rst:
```diff
@@ -0,0 +1,152 @@
+==============
+
+Some users say ``.hello`` out loud, and others will say it with an action. How
+do you react to these? Let's go back to the example of the ``shrug`` command:
+it uses a second decorator, ``action_command('shrugs')``, with a different
+name. How does that work?
```
At first I thought, "We should link back to the example above."

Then I realized there would be no clear way for the reader to scroll back down. It wouldn't be ideal for navigation. But the example is so short, we might as well just repeat it for clarity.

```suggestion
do you react to these? Let's go back to the example of the ``shrug`` command::

    @plugin.command('shrug')
    @plugin.action_command('shrugs')
    def shrug(bot, trigger):
        bot.say('¯\\_(ツ)_/¯')

Notice that it also uses a second decorator, ``action_command('shrugs')``,
with a different name. How does that work?
```

dgw commented on docs/source/plugin/tutorials/playing-with-commands.rst:
```diff
@@ -0,0 +1,152 @@
+
+We also saw that you have more ways to trigger a callable, and you can read
+more in :doc:`the plugin anatomy chapter </plugin/anatomy>` (see
+:ref:`plugin-anatomy-rules` in particular).
```
```suggestion
:ref:`how to define rules <plugin-anatomy-rules>`, in particular).
```

dgw reviewed (commented):
I took a different approach this time, reading through the rendered pages on our deploy preview. Really, there are only a couple of things I'd change now. I've tried to make it as easy as possible with suggestions you can apply through GitHub's GUI.

Meanwhile, I'll leave it to @SnoopJ to explain what he meant earlier today on IRC by saying the tutorials are a "big question mark", if he wants to.

Exirel commented:
> Really, there are only a couple of things I'd change now.

Great! I applied them all, so it's only a mater of a squash now.

> explain what he meant earlier today on IRC by saying the tutorials are a "big question mark"

IIRC it's something like "that's the only thing left to do". But I just added a PR for even more docs so...

dgw commented:
I say squash at your leisure. Don't know what's up with Coveralls check statuses getting stuck on Pending, but if we're lucky the squashed history will have enough of whatever it takes to get that to report correctly. (The earlier commits on this branch did, after all.)

Exirel commented:
Hey it worked for coverall! :grin:

dgw commented:
Indeed it did work for Coveralls. But you lost some of the changes in your squash 😩, _at least_ the ones that were just applied right before you did it.

I don't have a7cfd2828c75100c169eb718c5fa38281b158e5f locally, but I did have its parent commit, so I was able to reset the branch. I'll reapply the changes.

dgw reviewed (approved):
Oh yeah, I have to actually approve this now that it's squashed properly.

If you want to `--amend` the HEAD commit so it'll be verified/signed, feel free. Just please don't rebase again and lose more changes. 😜
